### ISSUE TITLE
Choosing the right epoch parameter

### ISSUE THREAD

**Issue by jwheatp** [lyst/lightfm#189]
Hi,

Firstly, thank you for your great work !

Could you give us more information / insights about how to intuitively set up the `epoch` parameter of the `fit` function ?

I understood in this issue https://github.com/lyst/lightfm/issues/183 that is was important. Indeed, in my script, changing this parameter from `30` to `300` make appears new items in the top-10 recommendations rank (when using the `predict` function).
Also after running several times the `predict` function with same parameters / inputs, the order of items in top-10 of recommendations changes a lot. (For example, a item can appear in the 6th position, and another time in the 3rd position). Does it mean the model has not trained enough ?***

My dataset shape is : 80000 users and 15000 items

Thanks in advance for your answers,

Julien

---
***Update : I actually changed the parameter `no_components` from `10` to `30` and the top-ranking is now less changing when I run several times my script.

**Comment by maciejkula** [comment]
1. LightFM is a stochastic gradient descent model, and so the model parameters will vary slightly from run to run, unless you fix the [random seed](https://github.com/lyst/lightfm/blob/master/lightfm/lightfm.py#L52)
2. If you're seeing large variations, chances are something is wrong with your model.

To select the right values, I would define an evaluation function and run a hyperparameter search, for example using sklearn's [ParameterSampler](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.ParameterSampler.html#sklearn.model_selection.ParameterSampler).

**Comment by jwheatp** [comment]
Thank you for your fast answer !
